### PR TITLE
[bitnami/kuberay] feat: :tada: Add chart

### PR DESCRIPTION
**Pull request by javsalgar**
### Description of the change

This PR adds kuberay based on the upstream kuberay charts. In this PR we mix the three charts (operator, apiserver and cluster) so it can all be deployed using the same chart. Features all Bitnami standards.

### Applicable issues

- fixes #

### Checklist

- [ ] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [ ] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [ ] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [ ] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/kuberay/README.md [hunk 1]
```diff
@@ -0,0 +1,562 @@
...
+
+# KubeRay packaged by Bitnami
+
+KubeRay is a Kubernetes operator for deploying and management of Ray applications on Kubernetes using CustomResourceDefinitions.
+
+[Overview of KubeRay](https://www.ray.io/)
+
+Trademarks: This software listing is packaged by Bitnami. The respective trademarks mentioned in the offering are owned by the respective companies, and use of them does not imply any affiliation or endorsement.
+
+## TL;DR
```
Comment (carrodher) [inline bitnami/kuberay/README.md:11]: It seems this README is not using the latest changes implemented by @jotamartos at https://github.com/bitnami/charts/pull/21009. From that moment, we should follow that structure for all existing and new Helm charts

File: bitnami/kuberay/crds/ray.io_rayclusters.yaml [hunk 2]
```diff
@@ -0,0 +1,13370 @@
+
```
Comment (migruiz4) [inline bitnami/kuberay/crds/ray.io_rayclusters.yaml:1]: I personally like adding a comment at the top of CRDs files with the source, e.g.:
```
# https://github.com/ray-project/kuberay/blob/v1.0.0/ray-operator/config/crd/bases/ray.io_rayclusters.yaml
```

This is especially helpful to detect outdated CRDs and quickly update them. WDYT?

**Review by migruiz4 — COMMENTED** [review]
The following values are referenced in the templates, but missing in the values.yaml:
```
operator.existingConfigMap
operator.enableBatchScheduler
apiserver.watchNamespaces
operator.watchNamespaces
apiserver.watchAllNamespaces
operator.rbac.allNamespaces
operator.watchAllNamespaces
apiserver.rbac.allNamespaces
```

File: bitnami/kuberay/README.md (hunk 1 above)
Comment (javsalgar) [reply bitnami/kuberay/README.md:11]: Fixed

File: bitnami/kuberay/crds/ray.io_rayclusters.yaml (hunk 2 above)
Comment (javsalgar) [reply bitnami/kuberay/crds/ray.io_rayclusters.yaml:1]: Done

File: bitnami/kuberay/templates/_helpers.tpl [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+{{- define "kuberay.cluster.fullname" -}}
+{{- printf "%s-%s" (include "common.names.fullname" .) "cluster" | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+
+{{/*
+Create the name of the service account to use (Kuberay Operator)
+*/}}
+{{- define "kuberay.cluster.serviceAccountName" -}}
+{{- if .Values.cluster.serviceAccount.create -}}
+    {{ default (include "kuberay.cluster.fullname" .) .Values.operator.serviceAccount.name }}
```
[the remark this answers is not in the document]
Comment (javsalgar) [reply bitnami/kuberay/templates/_helpers.tpl:108]: Good catch!

File: bitnami/kuberay/templates/cluster/service-account.yaml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  namespace: {{ .Release.Namespace | quote }}
+  {{- $versionLabel := dict "app.kubernetes.io/version" ( include "common.images.version" ( dict "imageRoot" .Values.operator.image "chart" .Chart ) ) }}
+  {{- $labels := include "common.tplvalues.merge" ( dict "values" ( list .Values.commonLabels $versionLabel ) "context" . ) }}
+  labels: {{- include "common.labels.standard" ( dict "customLabels" $labels "context" $ ) | nindent 4 }}
+  {{- if or .Values.operator.serviceAccount.annotations .Values.commonAnnotations }}
+  {{- $annotations := include "common.tplvalues.merge" ( dict "values" ( list .Values.operator.serviceAccount.annotations .Values.commonAnnotations ) "context" . ) }}
+  annotations: {{- include "common.tplvalues.render" ( dict "value" $annotations "context" $) | nindent 4 }}
+  {{- end }}
+automountServiceAccountToken: {{ .Values.operator.serviceAccount.automountServiceAccountToken }}
+{{- end }}
```
[the remark this answers is not in the document]
Comment (javsalgar) [reply bitnami/kuberay/templates/cluster/service-account.yaml:20]: Fixed!

File: bitnami/kuberay/values.yaml [hunk 5]
```diff
@@ -0,0 +1,1624 @@
...
+    ##
+    tolerations: []
+    ## @param cluster.head.updateStrategy.type Ray Cluster Worker (common) statefulset strategy type
+    ## ref: https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#update-strategies
+    ##
+    updateStrategy:
+      ## StrategyType
+      ## Can be set to RollingUpdate or OnDelete
+      ##
+      type: RollingUpdate
```
[the remark this answers is not in the document]
Comment (javsalgar) [reply bitnami/kuberay/values.yaml:1365]: Removing it as it does not exist in RayClusters

File: bitnami/kuberay/values.yaml [hunk 6]
```diff
@@ -0,0 +1,1624 @@
...
+      replicaCount: 1
+      ## @param cluster.worker.common.customLivenessProbe Custom livenessProbe that overrides the default one
+      ##
+      customLivenessProbe: {}
+      ## @param cluster.worker.common.customReadinessProbe Custom readinessProbe that overrides the default one
+      ##
+      customReadinessProbe: {}
+      ## @param cluster.worker.common.customStartupProbe Custom startupProbe that overrides the default one
+      ##
+      customStartupProbe: {}
```
[the remark this answers is not in the document]
Comment (javsalgar) [reply bitnami/kuberay/values.yaml:1443]: Fixed

File: bitnami/kuberay/values.yaml [hunk 7]
```diff
@@ -0,0 +1,1624 @@
...
+  serviceType: LoadBalancer
+  ## @section Ray Cluster Head Parameters
+  ##
+  head:
+    ## @param cluster.head.rayStartParams Set Ray start parameters
+    ##
+    rayStartParams: {}
+    ## @param cluster.head.replicaCount Number of Ray Cluster Worker (common) replicas to deploy
+    ##
+    replicaCount: 1
```
[the remark this answers is not in the document]
Comment (javsalgar) [reply bitnami/kuberay/values.yaml:1263]: Removing it as it is not part of the RayCluster head spec

File: bitnami/kuberay/values.yaml [hunk 8]
```diff
@@ -0,0 +1,1624 @@
...
+  ## @section Kuberay API Server Metrics Parameters
+  ##
+
+  ## Prometheus metrics
+  ##
+  metrics:
+    ## @param apiserver.metrics.enabled Enable the export of Prometheus metrics
+    ##
+    enabled: false
+    ## @param apiserver.metrics.annotations [object] Annotations for the server service in order to scrape metrics
```
[the remark this answers is not in the document]
Comment (javsalgar) [reply bitnami/kuberay/values.yaml:1195]: Fixed

File: bitnami/kuberay/values.yaml [hunk 9]
```diff
@@ -0,0 +1,1624 @@
...
+  ## Prometheus metrics
+  ##
+  metrics:
+    ## @param operator.metrics.enabled Enable the export of Prometheus metrics
+    ##
+    enabled: false
+    ## @param operator.metrics.annotations [object] Annotations for the server service in order to scrape metrics
+    annotations:
+      prometheus.io/scrape: "true"
+      prometheus.io/port: "{{ .Values.server.service.ports.metrics }}"
```
[the remark this answers is not in the document]
Comment (javsalgar) [reply bitnami/kuberay/values.yaml:623]: Fixed

File: bitnami/kuberay/values.yaml [hunk 10]
```diff
@@ -0,0 +1,1624 @@
...
+    ## @param operator.service.sessionAffinityConfig Additional settings for the sessionAffinity
+    ## sessionAffinityConfig:
+    ##   webIP:
+    ##     timeoutSeconds: 300
+    ##
+    sessionAffinityConfig: {}
+
+  ## ref: http://kubernetes.io/docs/user-guide/ingress/
+  ##
+  ingress:
```
[the remark this answers is not in the document]
Comment (javsalgar) [reply bitnami/kuberay/values.yaml:470]: Added it!

**Review by migruiz4 — APPROVED** [review]
Thank you for applying the suggestions, looks good to me!

**Pull request merged**